W-18564545 intro to mcp and a2a with flex (#845)
mulesoft/docs-gateway: gateway/1.9/modules/ROOT/pages/flex-agent-secure.adoc

= Securing Agent Interactions with Flex Gateway
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]
:imagesdir: ../assets/images

> Flex Gateway supports both the Model Context Protocol (MCP) and the Agent2Agent (A2A) Protocol, enabling you to protect your agents and systems. Using Flex Gateway with A2A and MCP, you can enforce policies across agent connections and monitor agent activities for enhanced visibility. This comprehensive approach provides centralized oversight, agent visibility, logging, and valuable insights, ensuring effective governance and security for your agent-based architecture.

Reviewer: ```suggestion
Flex Gateway supports both the Model Context Protocol (MCP) and the Agent2Agent (A2A) Protocol, enabling you to protect your agents and systems.  Flex Gateway enables you to enforce policies across agent connections and monitor agent activities for enhanced visibility. Flex Gateway provides centralized oversight, agent visibility, logging, and valuable insights, ensuring effective governance and security for your agent-based architecture.
```

== Flex Gateway A2A Support
A2A is an open standard that defines how agents interact with each other, enabling interoperability for agents built by different teams, using different technologies, and hosted by different organizations. 

> Using Flex Gateway with A2A, you can protect agent interactions by requiring appropriate authentication and authorization for agent requests. You can also enhance agent requests by modifying incoming prompts with additional context to improve server agent execution. 

Reviewer: ```suggestion
Flex Gateway protects agent A2A interactions by requiring appropriate authentication and authorization for agent requests. You can enhance agent requests by modifying incoming prompts with additional context to improve server agent execution. 
```

== Flex Gateway MCP Support
MCP is an open protocol that defines how agents interact with tools or external context, enabling agents to connect with the data and tools they need. 

> Using Flex Gateway with MCP, you can secure connections by restricting MCP endpoint access to authorized agents only, enforce rate limits to help your organization meet performance requirements, and simplify governance by providing centralized visibility and control over all interactions.

Reviewer: ```suggestion
Flex Gateway secures MCP connections by restricting MCP endpoint access to authorized agents only, enforcing rate limits to help your organization meet performance requirements, and simplify governance by providing centralized visibility and control over all interactions.
```

== Use Flex Gateway A2A and MCP Policies

Flex Gateway A2A and MCP policies provide enhanced security and control over agent interactions. For A2A, you can proxy agents by rewriting Agent Cards, log or block sensitive information in messages, modify prompt behavior, and use server-sent events (SSE) content logging for audit and compliance needs. For MCP, you can use attribute-based access control to manage access to MCP servers. 

> You can use most Flex Gateway included policies to secure and manage MCP and A2A connections. Use policies such as xref:policies-included-rate-limiting.adoc[] or xref:policies-included-spike-control.adoc[] to manage the number of requests to MCP and A2A servers. Use the Message Logging policy to monitor agent interactions. All Flex Gateway included policies work with MCP and A2A, except for the xref:policies-included-http-caching.adoc[] and the xref:policies-included-schema-validation.adoc[].

Reviewer: ```suggestion
You can use Flex Gateway included policies to secure and manage MCP and A2A connections. Use policies such as xref:policies-included-rate-limiting.adoc[] or xref:policies-included-spike-control.adoc[] to manage the number of requests to MCP and A2A servers. Use the Message Logging policy to monitor agent interactions.
```

We should add notes to the policy themselves, rather than here.

Reviewer: Listed this as a to do for me in the content plan.
